src/java/org/apache/cassandra/index/sai/disk/PrimaryKeyWithSource.java

```diff
@@ -103,4 +103,28 @@ public int compareTo(PrimaryKey o)
         }
         return primaryKey.compareTo(o);
     }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (o instanceof PrimaryKeyWithSource)
+        {
+            var other = (PrimaryKeyWithSource) o;
+            if (sourceSstableId.equals(other.sourceSstableId))
+                return sourceRowId == other.sourceRowId;
+        }
+        return primaryKey.equals(o);
+    }
+
+    @Override
+    public int hashCode()
```

Comment:
let's fix the PK implementations to not call loadDeferred() in hashCode() while we're at it -- hashing the token should be just fine

Reply:
@mike-tr-adamson asked me to include more than the `token` here: https://github.com/datastax/cassandra/pull/725#discussion_r1317370340

In practice, I we don't hash these all that often.

Reply:
Surely hashing the token isn't enough for wide rows?

Reply:
I updated the implementation to throw an exception. Let's see if any tests fail.

Reply:
Looks like several tests failed. I'll look closer to see if we need hashCode for the underlying code.